plugins/janus_videoroom.c

```diff
@@ -3467,355 +3417,168 @@ static void *janus_videoroom_handler(void *data) {
-									h264_pt,						/* H264 payload type */
-									b,								/* Bandwidth */
-									video_mode,						/* The publisher gets a recvonly or inactive back */
-									h264_pt, 						/* H264 payload type */
-									h264_pt, 						/* H264 payload type */
-									h264_pt, 						/* H264 payload type */
-									h264_pt, 						/* H264 payload type */
-									h264_pt, 						/* H264 payload type */
-									h264_pt); 						/* H264 payload type */
-							}
-							break;
-						default:
-							/* Shouldn't happen */
-							break;
-					}
-				} else {
-					video_mline[0] = '\0';
-				}
-				if(data) {
-					g_snprintf(data_mline, 256, sdp_d_template);
-				} else {
-					data_mline[0] = '\0';
-				}
-				g_snprintf(sdp, 1280, sdp_template,
-					janus_get_real_time(),			/* We need current time here */
-					janus_get_real_time(),			/* We need current time here */
-					participant->room->room_name,	/* Video room name */
-					audio_mline,					/* Audio m-line, if any */
-					video_mline,					/* Video m-line, if any */
-					data_mline);					/* Data channel m-line, if any */
-
-				char *newsdp = g_strdup(sdp);
-				if(video && b == 0) {
-					/* Remove useless bandwidth attribute */
-					newsdp = janus_string_replace(newsdp, "b=AS:0\r\n", "");
-				}
-				/* Lock listeners_mutex to protect recorders from race conditions. */
-				janus_mutex_lock(&participant->listeners_mutex);
-				/* Is this room recorded? */
-				if(videoroom->record || participant->recording_active) {
-					char filename[255];
-					gint64 now = janus_get_real_time();
 					if(audio) {
-						memset(filename, 0, 255);
-						if(participant->recording_base) {
-							/* Use the filename and path we have been provided */
-							g_snprintf(filename, 255, "%s-audio", participant->recording_base);
-							participant->arc = janus_recorder_create(videoroom->rec_dir,
-								janus_videoroom_audiocodec_name(participant->room->acodec), filename);
-							if(participant->arc == NULL) {
-								JANUS_LOG(LOG_ERR, "Couldn't open an audio recording file for this publisher!\n");
-							}
-						} else {
-							/* Build a filename */
-							g_snprintf(filename, 255, "videoroom-%"SCNu64"-user-%"SCNu64"-%"SCNi64"-audio",
-								videoroom->room_id, participant->user_id, now);
-							participant->arc = janus_recorder_create(videoroom->rec_dir,
-								janus_videoroom_audiocodec_name(participant->room->acodec), filename);
-							if(participant->arc == NULL) {
-								JANUS_LOG(LOG_ERR, "Couldn't open an audio recording file for this publisher!\n");
-							}
+						switch(videoroom->acodec) {
+							case JANUS_VIDEOROOM_OPUS:
+								if(opus_pt < 0) {
+									if(pass == 1)
+										JANUS_LOG(LOG_WARN, "Videoroom is forcing OPUS, but publisher didn't offer any... rejecting audio\n");
+								} else {
+									int pt = (pass == 1 ? opus_pt : OPUS_PT);
+									g_snprintf(audio_mline, 256, sdp_a_template_opus,
```

Comment:
Why not make a function for audio formats as well, as you did for video? Even though the number of arguments there is always the same, it would provide a nice interface in case any of them requires more or less in the future. I'd add the audio codec enum as one of the arguments there as well.